src/ensRegistry.ts

```diff
@@ -1,73 +1,69 @@
 // Import types and APIs from graph-ts
 import {
-  Address,
-  Bytes,
   ByteArray,
   crypto,
+  ens,
 } from '@graphprotocol/graph-ts'
 
 // Import event types from the registry contract ABI
-import { NewOwner, Transfer as TransferEvent, NewResolver, NewTTL } from './types/ENSRegistry/EnsRegistry'
+import { NewOwner, Transfer, NewResolver, NewTTL } from './types/ENSRegistry/EnsRegistry'
 
 // Import entity types generated from the GraphQL schema
-import { Domain, Account } from './types/schema'
+import { Account, Domain, Resolver } from './types/schema'
 
 // Handler for NewOwner events
-export function newOwner(event: NewOwner): void {
-  let subnode = crypto.keccak256(concat(event.params.node, event.params.label)).toHex()
-
-  let account = new Account(event.params.owner.toHex())
+export function handleNewOwner(event: NewOwner): void {
+  let account = new Account(event.params.owner.toHexString())
   account.save()
 
+  let subnode = crypto.keccak256(concat(event.params.node, event.params.label)).toHexString()
   let domain = new Domain(subnode)
+
+  // Get label and node names
+  let label = ens.nameByHash(event.params.label.toHexString())
+  if (label != null) {
+    domain.labelName = label
+  }
+
   domain.owner = account.id
-  domain.parent = event.params.node.toHex()
+  domain.parent = event.params.node.toHexString()
   domain.labelhash = event.params.label
   domain.save()
-
 }
 
 // Handler for Transfer events
-export function transfer(event: TransferEvent): void {
-  let node = event.params.node.toHex()
-
-  // // Create transfer if it does not exists yet
-  // let transfer = store.get('Transfer', domainId) as Transfer | null
-  // if (transfer == null) {
-  //   transfer = new Transfer()
-  //   transfer.domain = domainId
-  //   transfer.owners = new Array<Bytes>()
-  // }
-  //
-  // // Add the new owner to the list of historical owners of the domain
-  // let owners = transfer.owners
-  // owners.push(event.params.owner)
-  // transfer.owners = owners
-
-  let account = new Account(event.params.owner.toHex())
+export function handleTransfer(event: Transfer): void {
+  let account = new Account(event.params.owner.toHexString())
   account.save()
 
   // Update the domain owner
+  let node = event.params.node.toHexString()
   let domain = new Domain(node)
   domain.owner = account.id
   domain.save()
 }
 
 // Handler for NewResolver events
-export function newResolver(event: NewResolver): void {
-  let node = event.params.node.toHex()
-
+export function handleNewResolver(event: NewResolver): void {
+  let node = event.params.node.toHexString()
   let domain = new Domain(node)
   domain.resolver = event.params.resolver
   domain.save()
+
+  let resolver = new Resolver(event.params.resolver.toHexString())
+  resolver.domain = event.params.node
+  resolver.resolverEvents = []
```

Comment:
Isn't this automatically populated (eg, not stored in the datastore)?

Reply:
I think you are right, its an old pattern that I have engrained in my head. I will try to remove it, I believe it will work